### PR TITLE
fix default store destroy

### PR DESCRIPTION
**Pull request by aishmita-aggarwal**
Fix before_destroy callback validate_not_default to restrict destroy of default store...
Optimize ensure_default_exists_and_is_unique...
Clear cache for default store after destroy...
Modify specs for store

### CODE REVIEW

**Comment by priyank-gupta** [comment]
:+1: 
@Mafi88 @damianlegawiec ?

File: core/app/models/spree/store.rb [hunk 1]
```diff
@@ -21,28 +22,37 @@ def self.current(domain = nil)
 
     def self.default
       Rails.cache.fetch("default_store") do
-        where(default: true).first_or_initialize
+        find_or_initialize_by(default: true)
       end
     end
 
+    def self.has_default?
+      where(default: true).any?
```
Comment (kimquy) [inline core/app/models/spree/store.rb]: How about `exists?(default: true)` ?

File: core/app/models/spree/store.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@ class Store < Spree::Base
     scope :by_url, lambda { |url| where("url like ?", "%#{url}%") }
 
     before_save :clear_cache
+    after_destroy :clear_cache
```
Comment (kimquy) [inline core/app/models/spree/store.rb]: Maybe use `after_commit`

Comment (damianlegawiec) [inline core/app/models/spree/store.rb]: @kimquy :+1: definitely `after_commit`

**Comment by damianlegawiec** [comment]
@aishmita-aggarwal this was fixed via https://github.com/spree/spree/pull/7223

**Pull request closed without merging**